### PR TITLE
Add optional binary flag to open-file-to-*

### PR DESCRIPTION
**Pull request by owaddell**
@burgerrg are you open to a change like this? If so, I'll update the documentation and tests.

We've found that folks omit `O_TRUNC` by mistake when they meant to open a binary file to replace. I botched this in script-testing.ss, for example. If the `open-file-to-*` convenience procedures gave us the option of binary I/O, it might help folks avoid this mistake.

### CODE REVIEW

**Comment by owaddell** [comment]
We're also considering the alternative of adding `open-binary-file-to-*` in place of the optional argument.

**Comment by burgerrg** [comment]
I like adding the ability to open a binary file and think the binary? flag is a reasonable option.

**Comment by burgerrg** [comment]
I also think it would be useful to pass an optional file mode to the open-file-* functions that need it, with a default of #o666 for non-Windows systems and #o777 for Windows.

**Comment by burgerrg** [comment]
Or maybe just change the default #o777 to #o666 on non-Windows systems and have the user call the complete open-file for other cases.

File: src/swish/io.ss [hunk 1]
```diff
@@ -689,32 +689,44 @@
...
+    (case-lambda
+     [(name) (open-file-to-read name #f)]
+     [(name binary?)
+      (open-file name O_RDONLY 0 (if binary? 'binary-input 'input))]))
+
+  (define open-file-to-write
+    (case-lambda
+     [(name) (open-file-to-write name #f)]
+     [(name binary?)
+      (open-file name (+ O_WRONLY O_CREAT O_EXCL) #o777
```
Comment (burgerrg) [inline src/swish/io.ss]: Use #o666 to be consistent with Chez Scheme. On Windows, it's ignored.

File: src/swish/io.ss [hunk 2]
```diff
@@ -689,32 +689,44 @@
...
+        (if binary? 'binary-output 'output))]))
+
+  (define open-file-to-append
+    (case-lambda
+     [(name) (open-file-to-append name #f)]
+     [(name binary?)
+      (open-file name (+ O_WRONLY O_CREAT O_APPEND) #o777
```
Comment (burgerrg) [inline src/swish/io.ss]: Use #o666 to be consistent with Chez Scheme. On Windows, it's ignored.

File: src/swish/io.ss [hunk 3]
```diff
@@ -689,32 +689,44 @@
...
+        (if binary? 'binary-append 'append))]))
+
+  (define open-file-to-replace
+    (case-lambda
+     [(name) (open-file-to-replace name #f)]
+     [(name binary?)
+      (open-file name (+ O_WRONLY O_CREAT O_TRUNC) #o777
```
Comment (burgerrg) [inline src/swish/io.ss]: Use #o666 to be consistent with Chez Scheme. On Windows, it's ignored.

File: src/swish/io.ss (hunk 1 above)
Comment (owaddell) [reply src/swish/io.ss]: Good idea.

File: src/swish/io.ss (hunk 2 above)
Comment (owaddell) [reply src/swish/io.ss]: Good idea.

File: src/swish/io.ss (hunk 3 above)
Comment (owaddell) [reply src/swish/io.ss]: Good idea.

**Comment by owaddell** [comment]
I opted for the `open-binary-file-to-*` variants that Chris suggested. It makes the intent clearer.

**Pull request merged**